dotnet tool restore interactive doesn't prompt for DeviceFlow since .NET 6 SDK

### Describe the bug

If I use .NET 5 SDK, and call `dotnet tool restore --interactive`, I get presented with the DeviceFlow authentication flow from NuGet's MicrosoftCredentialProvider plugin (https://github.com/microsoft/artifacts-credprovider).

I get something like this:

```
[CredentialProvider]DeviceFlow: https://XXX.pkgs.visualstudio.com/_packaging/XXX/nuget/v3/index.json
[CredentialProvider]ATTENTION: User interaction required.

**********************************************************************

To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code XXXXXXXX to authenticate.

**********************************************************************
```

However, if I need the same thing in .NET 6 SDK (using 6.0.100), the output is silent, I don't receive any prompt on the output.

I found a workaround to add `--verbosity minimal` in the command parameters. But without this, it seems that the default verbosity is `quiet` which doesn't include the interactive auth flow prompt.

In .NET 5, the verbosity switch didn't have any effect for `dotnet tool restore`, I even created issue #16535. but in .NET 6, suddenly, the verbosity is working. It's just that the default verbosity is too quiet and is missing the important prompts.

### To Reproduce

- ensure that the tool packages to be restored aren't in NuGet cache
- ensure that you use feed that needs MicrosoftCredentialProvider auth (https://github.com/microsoft/artifacts-credprovider), I'm using Azure DevOps Artifacts
- ensure you don't have any MicrosoftCredentialProvider token already in cache
- call `dotnet tool restore --interactive`
- **expected:** prompt for device login
- **actual:** no output, program keeps running and waiting for the auth

### Further technical details

```
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff
```

I've experienced this as well on .NET 7. 

Running `dotnet tool install -g Microsoft.dotnet-interactive` in a cmd prompt yields this error, reflecting a package source I have to an Azure DevOps artifacts feed:

>  Determining projects to restore...
C:\Program Files\dotnet\sdk\7.0.102\NuGet.targets(132,5): warning : The plugin credential provider could not acquire cr
edentials. Authentication may require manual action. Consider re-running the command with --interactive for `dotnet`, /
p:NuGetInteractive="true" for MSBuild or removing the -NonInteractive switch for `NuGet` [C:\Users\whit_\AppData\Local\
Temp\2f906456-9768-478a-9646-8be22db8893f\restore.csproj]
C:\Users\xyz123\AppData\Local\Temp\2f906456-9768-478a-9646-8be22db8893f\restore.csproj : error NU1301: Unable to load th
e service index for source https://pkgs.dev.azure.com/innovian/Empirica/_packaging/Empirica/nuget/v3/index.json.
  Failed to restore C:\Users\xyz123\AppData\Local\Temp\2f906456-9768-478a-9646-8be22db8893f\restore.csproj (in 8.27 sec)
  .
The tool package could not be restored.
Tool 'microsoft.dotnet-interactive' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool

Re-running it with `dotnet tool install -g Microsoft.dotnet-interactive --verbosity minimal --interactive` prompted me to visit https://microsoft.com/devicelogin and provide a code, which after logging in successfully completed the restore and installed the tool (and I later got an email that a personal access token has been issued on my behalf for the vso.drop_write scope.

I'm pretty sure something like this is the basis for the [issue ](https://github.com/microsoft/vscode-jupyter/issues/12586) I'm having with using .NET interactive notebooks in VS Code in that I cannot install the NuGet packages because it cannot authenticate to my private Artifact feeds.

I can confirm that this is still not working with .NET 8 at the end of March:

```
S:\Renfield\CloudSync>dotnet restore --interactive --verbosity minimal
  Determining projects to restore...
S:\Renfield\CloudSync\CloudSync.SharepointAdapter\CloudSync.SharepointAdapter.csproj : error NU1301: Unable to load the
 service index for source https://nuget.telerik.com/v3/index.json. [S:\Renfield\CloudSync\CloudSync.sln]
S:\Renfield\CloudSync\CloudSync\CloudSync.csproj : error NU1301: Unable to load the service index for source https://nu
get.telerik.com/v3/index.json. [S:\Renfield\CloudSync\CloudSync.sln]

```
I have installed the latest credential provider with the `iex "& { $(irm https://aka.ms/install-artifacts-credprovider.ps1) } -AddNetfx"` command.

--- a/src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs
+++ b/src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs
@@ -698,6 +698,7 @@ namespace Microsoft.DotNet.Cli.NuGetPackageDownloader
             try
             {
                 SourceRepository repository = GetSourceRepository(source);
+                DefaultCredentialServiceUtility.SetupDefaultCredentialService(new NuGetConsoleLogger(), !_restoreActionConfig.Interactive);
                 PackageMetadataResource resource = await repository
                     .GetResourceAsync<PackageMetadataResource>(cancellationToken).ConfigureAwait(false);
 

--- a/src/Cli/dotnet/commands/dotnet-tool/restore/ToolRestoreCommand.cs
+++ b/src/Cli/dotnet/commands/dotnet-tool/restore/ToolRestoreCommand.cs
@@ -63,7 +63,7 @@ namespace Microsoft.DotNet.Tools.Tool.Restore
             _configFilePath = result.GetValue(ToolRestoreCommandParser.ConfigOption);
             _sources = result.GetValue(ToolRestoreCommandParser.AddSourceOption);
             _verbosity = result.GetValue(ToolRestoreCommandParser.VerbosityOption);
-            if (!result.HasOption(ToolRestoreCommandParser.VerbosityOption))
+            if (!result.HasOption(ToolRestoreCommandParser.VerbosityOption) && result.GetValue(ToolCommandRestorePassThroughOptions.InteractiveRestoreOption))
             {
                 _verbosity = VerbosityOptions.minimal;
             }

--- a/src/Cli/dotnet/commands/dotnet-tool/restore/ToolRestoreCommand.cs
+++ b/src/Cli/dotnet/commands/dotnet-tool/restore/ToolRestoreCommand.cs
@@ -3,6 +3,7 @@
 
 using System.CommandLine;
 using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
 using Microsoft.DotNet.Cli.ToolPackage;
 using Microsoft.DotNet.Cli.Utils;
 using Microsoft.DotNet.ToolManifest;
@@ -24,6 +25,7 @@ namespace Microsoft.DotNet.Tools.Tool.Restore
         private readonly string[] _sources;
         private readonly IToolPackageDownloader _toolPackageDownloader;
         private readonly VerbosityOptions _verbosity;
+        private readonly RestoreActionConfig _restoreActionConfig;
 
         public ToolRestoreCommand(
             ParseResult result,
@@ -61,6 +63,15 @@ namespace Microsoft.DotNet.Tools.Tool.Restore
             _configFilePath = result.GetValue(ToolRestoreCommandParser.ConfigOption);
             _sources = result.GetValue(ToolRestoreCommandParser.AddSourceOption);
             _verbosity = result.GetValue(ToolRestoreCommandParser.VerbosityOption);
+            if (!result.HasOption(ToolRestoreCommandParser.VerbosityOption))
+            {
+                _verbosity = VerbosityOptions.minimal;
+            }
+
+            _restoreActionConfig = new RestoreActionConfig(DisableParallel: result.GetValue(ToolCommandRestorePassThroughOptions.DisableParallelOption),
+                NoCache: result.GetValue(ToolCommandRestorePassThroughOptions.NoCacheOption) || result.GetValue(ToolCommandRestorePassThroughOptions.NoHttpCacheOption),
+                IgnoreFailedSources: result.GetValue(ToolCommandRestorePassThroughOptions.IgnoreFailedSourcesOption),
+                Interactive: result.GetValue(ToolCommandRestorePassThroughOptions.InteractiveRestoreOption));
         }
 
         public override int Execute()
@@ -130,7 +141,11 @@ namespace Microsoft.DotNet.Tools.Tool.Restore
                             nugetConfig: configFile,
                             additionalFeeds: _sources,
                             rootConfigDirectory: package.FirstEffectDirectory),
-                        package.PackageId, verbosity: _verbosity, ToVersionRangeWithOnlyOneVersion(package.Version), targetFramework
+                        package.PackageId,
+                        verbosity: _verbosity,
+                        ToVersionRangeWithOnlyOneVersion(package.Version),
+                        targetFramework,
+                        restoreActionConfig: _restoreActionConfig
                         );
 
                 if (!ManifestCommandMatchesActualInPackage(package.CommandNames, [toolPackage.Command]))

